packages/dashboard-core-plugins/src/linker/Linker.tsx

```diff
@@ -153,6 +153,7 @@ export class Linker extends Component<LinkerProps, LinkerState> {
       this.handleFilterColumnSelect
     );
     eventHub.on(InputFilterEvent.COLUMNS_CHANGED, this.handleColumnsChanged);
+    eventHub.on(PanelEvent.CLOSE, this.handlePanelClosed);
```

Comment:
remove

Reply:
Uhhh this change is needed. Don't revert.

@dsmmcken The `PanelEvent.CLOSED` doesn't get fired in this case, as the `PanelEvent.CLOSE` event gets fired before the `PanelErrorBoundary` gets mounted, so it can't emit the `PanelEvent.CLOSED`

Reply:
Correct. `ConsolePanel` fires it to close these panels. They never emit a `.CLOSED` event because those panels are never actually mounted. So we're listening for the `.CLOSE` here as well to pick up when a panel would get closed before it's mounted.
Golden layout mounts the panels in order, so when it mounts the `ConsolePanel`, it emits all the `.CLOSE` events to close the panels from the previous session, those panels that haven't been initialized yet won't emit the `.CLOSED` event.
We could have layout initialization check if there's any panels that need to be closed and remove them from the config beforehand, then Linker would need to check on startup too if all panels still exist; we could re-emit the `.CLOSED` event in cases when `.CLOSE` occurs on a panel that isn't mounted yet, but that's a little complicated because `.CLOSED` expects a `glContainer` which may not still exist; or we can just listen for `.CLOSE` and `.CLOSED` in linker and fix that case.

Reply:
oh... that panelevent.close. I am an idiot yes.